public/app/core/components/PasswordStrength.tsx

```diff
@@ -1,36 +1,60 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 
 export interface Props {
   password: string;
 }
 
-export class PasswordStrength extends React.Component<Props, any> {
-  constructor(props) {
+export class PasswordStrength extends PureComponent<Props, any> {
+  constructor(props: Props) {
     super(props);
+
+    this.state = {
+      getScore(password: string) {
```

Comment:
This PR looks good except this, defining a function in the state object looks very unorthodox, have never seen it before. Also State is untyped so if you want to use state it should have a type. @dprokop have you seen this before? adding a function to the state?

Reply:
I don't think this is particularly unorthodox - state, like props, is just a regular collection of whatever you want, functions not being special - the alternative I can see is to lazy load the whole component using zxcvbn and replace that, passing `getScore` as a prop, but that's just moving the logic upwards to the consumer.

Reply:
I haven't seen this approach being used neither. Not sure what the benefit of having this method in state is compared to regular instance method, bu this feels flaky. I see this being driven only by the dynamic import in `componentDidMount`. For that I suggest creating a higher order component that would serve as module loader and deliver the loaded module or part of its API to the desired component (PasswordStrength in this case)